conn.go

```diff
@@ -66,6 +67,8 @@ type authCreds struct {
 	auth   []byte
 }
 
+// Conn is the connection object for a ZK client, it holds all the stuff one
+// needs to communicate with ZK server, all the properties should be private.
```

Comment:
I would simplify this a bit.   Something like

```
// Conn is the client connection and tracks all details for communication with the server.
```

Reply:
Point taken. Thanks, I'll learn to improve in the future. :)